File: Source Server/SourceQuest4.5/SqlDataProvider/SqlDataProvider.Data/BallInfo.cs
namespace SqlDataProvider.Data
{
	public class BallInfo
	{
		public int ActionType
		{
			get;
			set;
		}

		public int Amount
		{
			get;
			set;
		}

		public int AttackResponse
		{
			get;
			set;
		}

		public string BombPartical
		{
			get;
			set;
		}

		public string BombSound
		{
			get;
			set;
		}

		public string Crater
		{
			get;
			set;
		}

		public int Delay
		{
			get;
			set;
		}

		public int DragIndex
		{
			get;
			set;
		}

		public string FlyingPartical
		{
			get;
			set;
		}

		public bool HasTunnel
		{
			get;
			set;
		}

		public int ID
		{
			get;
			set;
		}

		public bool IsSpin
		{
			get;
			set;
		}

		public int Mass
		{
			get;
			set;
		}

		public string Name
		{
			get;
			set;
		}

		public double Power
		{
			get;
			set;
		}

		public int Radii
		{
			get;
			set;
		}

		public bool Shake
		{
			get;
			set;
		}

		public string ShootSound
		{
			get;
			set;
		}

		public int SpinV
		{
			get;
			set;
		}

		public double SpinVA
		{
			get;
			set;
		}

		public int Weight
		{
			get;
			set;
		}

		public int Wind
		{
			get;
			set;
		}

		public bool IsSpecial()
		{
			int d = ID;
			if (d <= 64)
			{
				if (d > 16)
				{
					if (d == 59 || d == 64)
					{
						return true;
					}
					return false;
				}
				switch (d)
				{
				case 2:
				case 4:
					return false;
				default:
					return false;
				case 1:
				case 3:
				case 5:
				case 16:
					break;
				}
			}
			else
			{
				if (d <= 98)
				{
					if (d == 97 || d == 98)
					{
						return true;
					}
					return false;
				}
				if (d != 110 && d != 117 && (uint)(d - 10001) > 22u)
				{
					return false;
				}
			}
			return true;
		}
	}
}

File: NewRequest/modelo/BallInfo.php
<?php

/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/**
 * Description of BallInfo
 *
 * @author JeffzAtaa
 */
class BallInfo {

    //put your code here
    private $ID, $Power, $Radii, $FlyingPartical, $BombPartical, $Crater, $AttackResponse, $IsSpin, $SpinV, $SpinVA, $Amout, $Win, $DragIndex, $Weight, $Shake, $ShootSound, $BombSound, $ActionType, $Mass, $BombType;

    function getID()
    {
        return $this->ID;
    }
    function getPower()
    {
        return $this->Power;
    }
    function getRadii()
    {
        return $this->Radii;
    }
    function getFlyingPartical()
    {
        return $this->FlyingPartical;
    }
    function getBombPartical()
    {
        return $this->BombPartical;
    }
    function getCrater()
    {
        return $this->Crater;
    }
    function getAttackResponse()
    {
        return $this->AttackResponse;
    }
    function getIsSpin()
    {
        return $this->IsSpin;
    }
    function getSpinV()
    {
        return $this->SpinV;
    }
    function getSpinVA()
    {
        return $this->SpinVA;
    }
    function getAmount()
    {
        return $this->Amount;
    }
    function getWind()
    {
        return $this->Wind;
    }
    function getDragIndex()
    {
        return $this->DragIndex;
    }
    function getWeight()
    {
        return $this->Weight;
    }
    function getShake()
    {
        return $this->Shake;
    }
    function getShootSound()
    {
        return $this->ShootSound;
    }
    function getBombSound()
    {
        return $this->BombSound;
    }
    function getActionType()
    {
        return $this->ActionType;
    }
    function getMass()
    {
        return $this->Mass;
    }
    function getBombType()
    {
        return $this->BombType;
    }
    function setID($ID)
    {
        $this->ID = $ID;
    }
    function setPower($Power)
    {
        $this->Power = $Power;
    }
    function setRadii($Radii)
    {
        $this->Radii = $Radii;
    }
    function setFlyingPartical($FlyingPartical)
    {
        $this->FlyingPartical = $FlyingPartical;
    }
    function setBombPartical($BombPartical)
    {
        $this->BombPartical = $BombPartical;
    }
    function setCrater($Crater)
    {
        $this->Crater = $Crater;
    }
    function setAttackResponse($AttackResponse)
    {
        $this->AttackResponse = $AttackResponse;
    }
    function setIsSpin($IsSpin)
    {
        $this->IsSpin = $IsSpin;
    }
    function setSpinV($SpinV)
    {
        $this->SpinV = $SpinV;
    }
    function setSpinVA($SpinVA)
    {
        $this->SpinVA = $SpinVA;
    }
    function setAmount($Amount)
    {
        $this->Amount = $Amount;
    }
    function setWind($Wind)
    {
        $this->Wind = $Wind;
    }
    function setDragIndex($DragIndex)
    {
        $this->DragIndex = $DragIndex;
    }
    function setWeight($Weight)
    {
        $this->Weight = $Weight;
    }
    function setShake($Shake)
    {
        $this->Shake = $Shake;
    }
    function setShootSound($ShootSound)
    {
        $this->ShootSound = $ShootSound;
    }
    function setBombSound($BombSound)
    {
        $this->BombSound = $BombSound;
    }
    function setActionType($ActionType)
    {
        $this->ActionType = $ActionType;
    }
    function setMass($Mass)
    {
        $this->Mass = $Mass;
    }
    function setBombType($BombType)
    {
        $this->BombType;
    }
}
